datafusion/functions/src/core/getfield.rs

```diff
@@ -106,6 +106,9 @@ impl ScalarUDFImpl for GetFieldFunc {
         };
         let access_schema = GetFieldAccessSchema::NamedStructField { name: name.clone() };
         let arg_dt = args[0].get_type(schema)?;
+        if arg_dt.is_null() {
+            return Ok(DataType::Null);
+        }
         access_schema
```

Comment:
I quickly fixed it before noticing that there was already a PR
https://github.com/apache/datafusion/commit/82530608d6737c8d1906250dbe1aab6f426d47f4

I think we can remove `GetFieldAccessSchema`, `get_accessed_field`.

Reply:
Thanks @jayzhan211  

I agree the change in https://github.com/apache/datafusion/commit/6cb45b5a9f2b53688a9754b1a7d06665ebd73a6d to inline things looks nice -- shall we make another PR?